### PR TITLE
feature(dev-env): Add vip export sql command

### PR DESCRIPTION
**Pull request by aswasif007**
## Description

In this PR, we are adding the following commands:
```bash
$ vip export --help
  Usage: vip export [options] [command]
  
  Commands:
    sql  Import SQL to your database from a file
  
  Options:
    -d, --debug    Activate debug output
    -h, --help     Output the help for the (sub)command
    -v, --version  Output the version number
  
  Examples:
    - Export SQL file from your site
    $ vip export sql @mysite.develop
```

and,

```bash
$ vip export sql --help
  Usage: vip export-sql [options] 
  
  Options:
    -a, --app      Specify the app
    -d, --debug    Activate debug output
    -e, --env      Specify the environment
    -h, --help     Output the help for the (sub)command
    -o, --output   Specify the location where you want to save the export file
    -v, --version  Output the version number
  
  Examples:
    - Export SQL file from your site and save it to the specified location
    $ vip export sql @mysite.develop --output=/home/user/export.sql.gz
```

The `vip export sql` command exports the latest backup and saves it in user's filesystem.

## Steps to Test

1. Check out PR.
1. Run `rm -rf dist && npm link`
1. Run `node ./dist/bin/vip export sql @6721.production`
1. It should store the export SQL file in the current working directory. If you specify the file location using `/home/..../test.sql.gz` it would save the file there.
2. If you are an Automattician and using proxy, you will need to add the `VIP_PROXY` environment variable.

### CODE REVIEW

File: src/bin/vip-export.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ */
+
+/**
+ * Internal dependencies
+ */
+import command from '../lib/cli/command';
+import { trackEvent } from '../lib/tracker';
+
+command()
+	.command( 'sql', 'Import SQL to your database from a file' )
```
Comment (saroshaga) [inline src/bin/vip-export.js]:
```suggestion
	.command( 'sql', 'Export the contents of your database to an SQL file' )
```

Comment (saroshaga) [reply src/bin/vip-export.js]: same below

**Comment by saroshaga** [comment]
Will this be released on the @next version first @aswasif007 ?

**Comment by aswasif007** [comment]
@saroshaga will do a dev release from this branch so that Lucas can test it out.

File: src/bin/vip-export-sql.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * Internal dependencies
+ */
+import command from '../lib/cli/command';
+import { ExportSQLCommand } from '../commands/export-sql';
+import { makeCommandTracker } from '../lib/tracker';
+
+
+const examples = [
+  {
+    usage: 'vip export sql @mysite.develop --output=/home/user/export.sql.gz',
```
Comment (saroshaga) [inline src/bin/vip-export-sql.js]: Can we add another example without the `--output` flag? 

https://a8c.slack.com/archives/CLKJLBFB8/p1684941954409949?thread_ts=1684941892.014789&cid=CLKJLBFB8

Comment (saroshaga) [reply src/bin/vip-export-sql.js]: nevermind, I think this would make sense if we support outputting to stdout

File: src/lib/cli/format.js [hunk 3]
```diff
@@ -206,3 +206,17 @@ export const formatSearchReplaceValues = ( values, message ) => {
...
 	return formattedOutput;
 };
+
+// Format bytes into kilobytes, megabytes, etc based on the size
+export const formatBytes = ( bytes, decimals = 2 ) => {
+	if ( 0 === bytes ) {
+		return '0 Bytes';
+	}
+
+	const k = 1024;
```
Comment (saroshaga) [inline src/lib/cli/format.js]: Can we please fix the lint issue here?

File: src/commands/export-sql.js [hunk 4]
```diff
@@ -298,26 +306,44 @@ export class ExportSQLCommand {
...
 			try {
 				await createExportJob( this.app.id, this.env.id, latestBackup.id );
 			} catch ( err ) {
+				// Todo: match error code instead of message substring
+				if ( err?.message.includes( 'Backup Copy already in progress' ) ) {
+					await this.track( 'export_job_already_running_error', { errorMessage: err?.message } );
+					exit.withError(
+						'There is an export job already running for this site: ' +
+						`https://dashboard.wpvip.com/apps/${ this.app.id }/${ this.env.uniqueLabel }/data/database/backups\n` +
+						'Currently, we allow only one export job per site. Please try again later.'
```
Comment (saroshaga) [inline src/commands/export-sql.js]:
```suggestion
						'Currently, we allow only one export job at a time, per site. Please try again later.'
```

**Review by saroshaga — APPROVED** [review]

**Pull request merged**